### PR TITLE
RELATED: BB-257 onFiredDrillEvent

### PR DESCRIPTION
**Pull request by BugsBunny338**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/2/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/667/ : FAILURE in 6m 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/8/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/674/ : FAILURE in 2m 40s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/9/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/675/ : FAILURE in 1m 06s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/10/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/676/ : FAILURE in 1m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/13/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/679/ : SUCCESS in 1m 05s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/17/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/683/ : SUCCESS in 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/19/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/685/ : SUCCESS in 1m 07s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/20/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/686/ : SUCCESS in 1m 00s

File: src/Table/TableVisualization.jsx [hunk 1]
```diff
@@ -54,6 +55,7 @@ export default class TableVisualization extends Component {
     static defaultProps = {
         afm: {},
         drillableItems: [],
+        onFiredDrillEvent: noop,
```
Comment (martinmosko) [inline src/Table/TableVisualization.jsx]: `noop` vs `() => {}`

File: src/Chart/highcharts/commonConfiguration.js [hunk 2]
```diff
@@ -126,16 +126,16 @@ const BASE_TEMPLATE = {
     }
 };
 
-const registerDrilldownHandler = (configuration, chartOptions, afm) => {
+const registerDrilldownHandler = (configuration, chartOptions, drillConfig) => {
```
Comment (martinmosko) [inline src/Chart/highcharts/commonConfiguration.js]: https://confluence.intgdc.com/display/~vaclav.bohac/Code+Review+-+Checklist
unnecessary arrow functions ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/21/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/687/ : SUCCESS in 3m 00s

File: src/Chart/LineFamilyChartTransformation.jsx [hunk 3]
```diff
@@ -50,7 +49,9 @@ export default class LineFamilyChartTransformation extends Component {
             headers: PropTypes.arrayOf(PropTypes.object),
             rawData: PropTypes.arrayOf(PropTypes.array)
         }).isRequired,
+        afm: PropTypes.object,
         drillableItems: PropTypes.arrayOf(PropTypes.shape(DrillableItem)),
+        onFiredDrillEvent: PropTypes.func,
```
Comment (tomassatinsky) [inline src/Chart/LineFamilyChartTransformation.jsx]: Just a detail - I would prefer `onDrillEvent` as nothing may be fired if the callback returned `false`

Comment (tomassatinsky) [inline src/Chart/LineFamilyChartTransformation.jsx]: Just a detail - I would prefer `onDrillEvent` as nothing may be fired if the callback returned `false`

**Comment by tomassatinsky** [comment]
As this drill stuff bubbles through multiple files and none of it is being tested, then at least the test coffee integration tests would be worth adding at some point soon.

File: src/Chart/LineFamilyChartTransformation.jsx (hunk 3 above)
Comment (BugsBunny338) [reply src/Chart/LineFamilyChartTransformation.jsx]: Good point...

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/25/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/691/ : SUCCESS in 2m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-opensource-repository-check/29/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/695/ : SUCCESS in 1m 03s

File: src/Table/TableVisualization.jsx (hunk 1 above)
Comment (BugsBunny338) [reply src/Table/TableVisualization.jsx]: no strong opinion here ... it's the same, we use both, this particular file uses `noop`, so let's use `noop`

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-indigo-visualizations-merge-compliance-check/76/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-indigo-visualizations-unit-tests-zuul-el7/696/ : SUCCESS in 1m 00s

**Pull request merged**